beacon-chain/rpc/validator_server.go

```diff
@@ -139,14 +139,11 @@ func (vs *ValidatorServer) CommitteeAssignment(ctx context.Context, req *pb.Assi
 		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
 	}
 
-	// Advance state with empty transitions if the head state slot is farther
-	// than 1 epoch away from the request.
-	if req.EpochStart > 1 && s.Slot < helpers.StartSlot(req.EpochStart-1) {
-		slotsToAdvance := helpers.StartSlot(req.EpochStart - 1)
-		s, err = state.ProcessSlots(ctx, s, slotsToAdvance)
-		if err != nil {
-			return nil, err
-		}
+	// Advance state with empty transitions up to the requested slot.
+	slotsToAdvance := req.EpochStart * params.BeaconConfig().SlotsPerEpoch
```

Comment:
Any regression test that could be added? If that's a bigger task, please create an issue to track it

Reply:
A test like the following can be added, just not sure how valuable it is:

1. Current state slot at X, request slot at Y
2. Verify computed assignment Z is equal to Z (hardcoded test value)